scripts/ci/check_compliance.py

```diff
@@ -262,6 +263,38 @@ def run(self):
         for file in path.glob("**/board.yml"):
             self.check_board_file(file, vendor_prefixes)
 
+
+class ClangFormatCheck(ComplianceTest):
+    """
+    Check if clang-format reports any issues
+    """
+    name = "ClangFormat"
+    doc = "See https://docs.zephyrproject.org/latest/contribute/guidelines.html#clang-format for more details."
+    path_hint = "<git-top>"
+
+    def run(self):
+        for file in get_files():
+            diff = subprocess.Popen(('git', 'diff', '-U0', '--no-color', COMMIT_RANGE, '--', file),
+                                    stdout=subprocess.PIPE,
+                                    cwd=GIT_TOP)
+            try:
+                subprocess.run(('clang-format-diff.py', '-p1'),
+                               check=True,
+                               stdin=diff.stdout,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT,
+                               cwd=GIT_TOP)
+
+            except subprocess.CalledProcessError as ex:
+                patchset = unidiff.PatchSet.from_string(ex.output, encoding="utf-8")
+                for patch in patchset:
+                    for hunk in patch:
+                        # Set the annotation to the last line of the hunk
+                        line = hunk.source_start + hunk.source_length - 1
+                        self.fmtd_failure("notice", "ClangFormat", file, line=line,
+                                          desc=f'\r\n{hunk}')
```

Comment:
I was thinking something along these lines (LOL)?  
```suggestion
                for patch in patchset:
                    for hunk in patch:
                        for line in hunk:
                            # Set the annotation to the first offending line in the hunk
                            if line.is_removed:
                                self.fmtd_failure("notice", "ClangFormat", file, line=line.source_line_no,
                                                desc=f'\r\n{hunk}')
                                break
```